src/telemetry/Manager.h

```diff
@@ -50,6 +59,112 @@ class Manager
 	 */
 	virtual void InitPostScript();
 
+	enum class MetricType
+		{
+		Counter,
+		Gauge,
+		Histogram
+		};
+
+	/**
+	 * Captures information about counter and gauge metrics and their values.
+	 */
+	class CollectedValueMetric
+		{
+	public:
+		static CollectedValueMetric NewCounter(const broker::telemetry::metric_family_hdl* family,
+		                                       std::vector<std::string_view> label_values,
+		                                       std::variant<double, int64_t> value)
+			{
+			return CollectedValueMetric(MetricType::Counter, family, std::move(label_values),
+			                            value);
+			}
+
+		static CollectedValueMetric NewGauge(const broker::telemetry::metric_family_hdl* family,
+		                                     std::vector<std::string_view> label_values,
+		                                     std::variant<double, int64_t> value)
+			{
+			return CollectedValueMetric(MetricType::Gauge, family, std::move(label_values), value);
+			}
```

Comment:
Do these wrappers add much over users of the class calling the constructors themselves with the appropriate `MetricType` enum? Feels like it's just making the class API more complex for little gain, unless I'm missing something?

Reply:
Motivation was to protect a CollectValueMetric with MetricType::Histogram, but removed them in favor of less code now.